contrib/babelfishpg_tsql/src/collation.c

```diff
@@ -249,14 +267,46 @@ transform_funcexpr(Node* node)
  *		 col LIKE PATTERN -> col ILIKE PATTERN
  */
 static Node*
-transform_likenode(Node* node)
+transform_likenode(Node* node, Oid opno)
 {
 	if (node && IsA(node, OpExpr))
 	{
 		OpExpr	 *op = (OpExpr *) node;
 		like_ilike_info_t like_entry = tsql_lookup_like_ilike_table_internal(op->opno);
 		coll_info_t coll_info_of_inputcollid = tsql_lookup_collation_table_internal(op->inputcollid);
 
+		/*
+		 * We do not allow CREATE TABLE statements with CHECK constraint where the
+		 * constraint has an ILIKE operator and the collation is ci_as. But during
+		 * dump and restore, this kind of a table definition may be generated. At
+		 * this point we know that any tables being restored that match this pattern
+		 * are generated by pg_dump, and not created by a user. So, it is safe to go
+		 * ahead with replacing the ci_as collation with a corresponding cs_as one
+		 * if an ILIKE node is found during dump and restore. 
+		 */
+		init_and_check_collation_callbacks();
+		if ((*collation_callbacks_ptr->expr_contains_ilike_and_ci_as_coll)(node, false)
+				 && babelfish_dump_restore)
+		{
+			int		 collidx_of_cs_as;
+			
+			if (coll_info_of_inputcollid.collname)
+			{
+				collidx_of_cs_as =
+					tsql_find_cs_as_collation_internal(
+						tsql_find_collation_internal(coll_info_of_inputcollid.collname));
+				op->inputcollid = tsql_get_oid_from_collidx(collidx_of_cs_as);
+			}
+			else
+			{
+				/* If a collation is not specified, use the default one */
+				op->inputcollid = DEFAULT_COLLATION_OID;
+			}
```

Comment:
Do we have a test case for `else` block?

Reply:
We don't have a test case for this because this check is for psql syntax (similar to the test case that you asked about in hooks.c). TSQL syntax will not trigger this code block at all.

I have done manual testing for this in psql.

Reply:
Added tests in .mix files